Create JB blog (squashed)

mkondratek opened the pull request.

mkondratek committed:
Create JB blog (squashed)

mkondratek committed:
Create VL blog (squashed)

mkondratek committed:
Remove first, vcs log referencing paragraph of intro

mkondratek committed:
Apply review from @karolinavl

mkondratek committed:
Add git machete logo

mkondratek committed:
Attempt to adjust git machete logo size

mkondratek committed:
Extract logo from title

mkondratek committed:
Inline logo b/w intro paragraphs

PawelLipski commented on blog.md:
```diff
@@ -14,7 +18,18 @@ their relationship with each other and with the remote repository.
+## Git Machete
+
+The Git Machete plugin is a tool that helps developers handle multiple branches.
+The main features of the plugin are:
```
I wouldn't call the below `features`, rather more high-level `objectives`? `aims`?...

PawelLipski replied:
Also, `s/are/include` sounds better

PawelLipski commented on blog.md:
```diff
@@ -14,7 +18,18 @@ their relationship with each other and with the remote repository.
+- gives a bird's eye view of the repo
+- automates common flows...
+
+#### Git Machete Status Graph
```
Hmm fourth-level header under the second-level one?

PawelLipski commented on blog.md:
```diff
@@ -89,7 +104,44 @@ The final repository status is shown below.
+We are doing our best to provide new features and quickly react to the users' feedback.
+We are responding to the new issues and ideas.
+
+The feature that we are definitely aiming for now is the traverse functionality
```
```suggestion
The feature that we are definitely aiming for now is the semi-automatic traversal functionality
```

PawelLipski commented on blog.md:
```diff
@@ -89,7 +104,44 @@ The final repository status is shown below.
+
+To be more precise and understand the technical background
+we should look closer to the consequences of big and small PRs.
+**Big PRs are bad.** Reviewers are unwilling to even begin the review.
```
```suggestion
**Big PRs are bad,** especially if they mix refactors with bugfixes and new features.
Reviewers are less willing to even begin the review.
```

PawelLipski commented on blog.md:
```diff
@@ -89,7 +104,44 @@ The final repository status is shown below.
-### Try it now
+### Motivation
+
+The idea standing behind the plugin is automating the developer's work.
+We (developers) should strive for focusing on actual development
```
`developX` repeated  3 times :c

PawelLipski commented on blog.md:
```diff
@@ -14,7 +18,18 @@ their relationship with each other and with the remote repository.
+The main features of the plugin are:
+- helps keep branches up to date even if (side effects of small PRs):
+  - there many of them
+  - they're occasionally stacked on one another
```
```suggestion
  - they're occasionally stacked on one another, e.g. `develop` (arrow) `refactor/foo` (arrow) `bugfix/bar`
```

PawelLipski commented on blog.md:
```diff
@@ -1,7 +1,11 @@
+The version control system provided with the JetBrains's IDE products
+allows performing pretty much all everyday VCS actions directly via the IDE.
+Among the many useful benefits of the provided IDE and VCS integration,
+the full support of GitHub Pull Requests introduced in IDEA 2020.2 must be emphasized.
```
`s/emphasized/highlighted`? :thinking:  The sentence sounds too stiff/formal

mkondratek committed:
Fix headers (level them up)

mkondratek committed:
Apply review fixes (deformalize & cleanup)

Git Machete
- s/features/objectives
- add stacked PR example to objectives

Motivation
- resolve develop repetition
- point out big PRs weakness (refactor/bugfix/feature mix)

karolinavl commented on blog.md:
```diff
@@ -89,7 +104,45 @@ The final repository status is shown below.
+**Big PRs are bad,** especially if they mix refactors with bugfixes and new features.
+Reviewers are unwilling to even begin the review.
+They are more likely to skip vital parts of the newly-added code.
+It's hard/impossible to keep entire context in memory during review.
```
```suggestion
It's hard/impossible to keep the entire context in memory during a review.
```

karolinavl commented on blog.md:
```diff
@@ -89,7 +104,45 @@ The final repository status is shown below.
+
+Git Machete plugin is being developed and maintained continuously.
+We are doing our best to provide new features and quickly react to the users' feedback.
+We are responding to the new issues and ideas.
```
```suggestion
We are responding to new issues and ideas.
```

karolinavl commented on blog.md:
```diff
@@ -89,7 +104,45 @@ The final repository status is shown below.
+## Learn more
+
+The plugin is available from the [Marketplace](https://plugins.jetbrains.com/plugin/14221-git-machete "JetBrains Marketplace - Git Machete Plugin").
+It may differ a bit from presented version as some improvements may had been provided with the newest release.
```
```suggestion
It may differ a bit from the presented version as some improvements may have been provided with the newest release.
```

karolinavl commented on blog.md:
```diff
@@ -89,7 +104,45 @@ The final repository status is shown below.
+
+Creating and maintaining **small, focused, easy-to-review PRs** is a comfortable and desired approach
+for both the authors and, especially, the reviewers.
+This is exactly what Git Machete is promoting.
```
```suggestion
That is exactly what Git Machete is promoting.
```

mkondratek committed:
Apply review from @karolinavl
